### PR TITLE
Add the ability to cut out a Rectangle from ImageryLayers

### PR DESCRIPTION
**Pull request by likangning93**
This PR adds the ability to cut out a `Rectangle` from `ImageryLayers`.

imagery cutout

A caveat that I feel needs defense: this doesn't influence imagery tile selection. The intent was for this `Rectangle` to be dynamic, and it looked to me like getting this to function with the existing imagery tile selection code would have been pretty complicated. Let me know if I was wrong and I should fix that.

Besides this just being kind of a fun feature, we likely need this to implement https://github.com/AnalyticalGraphicsInc/cesium/issues/6531.

Client-side mosaics and client-side imagery reprojection, both of which need to preserve pixel detail, will  probably require doing something like generating high detail imagery for small areas on-the-fly and drawing it over low-detail "placeholder" imagery that covers the entire mosaic area.
The layers can be transparent, so layering them directly on top of each other isn't acceptable.

### CODE REVIEW

**Comment by bagnell** [comment]
@likangning93 This looks good to me. Can you merge in master?

**Comment by likangning93** [comment]
@bagnell merged in master

File: Specs/Scene/GlobeSurfaceTileProviderSpec.js [hunk 1]
```diff
@@ -573,13 +573,13 @@ defineSuite([
         switchViewMode(SceneMode.SCENE3D, new GeographicProjection(Ellipsoid.WGS84));
 
         return updateUntilDone(scene.globe).then(function() {
-            expect(scene).toRender([0, 0, 127, 255]); // default baseColor
+            expect(scene).toRender([0, 0, 128, 255]); // default baseColor
```
Comment (bagnell) [inline Specs/Scene/GlobeSurfaceTileProviderSpec.js]: This test is failing for me. The result is 127, not 128.

Comment (likangning93) [reply Specs/Scene/GlobeSurfaceTileProviderSpec.js]: Hmmm, I swear it was passing on Windows/Linux on my Intel system, but now I'm seeing the same thing on Windows + Nvidia. Either way, poorly designed spec...

**Comment by likangning93** [comment]
@bagnell this is up-to-date!

**Pull request merged**